tests/probes_test.go

```diff
@@ -257,11 +255,11 @@ func createTCPProbe(period int32, initialSeconds int32, port int) *v12.Probe {
 	return createProbeSpecification(period, initialSeconds, httpHandler)
 }
 
-func patchProbeWithIPv6Addr(existingProbe *v12.Probe, ipv6HostIP string) *v12.Probe {
+func patchProbeWithIPAddr(existingProbe *v12.Probe, ipHostIP string) *v12.Probe {
```

Comment:
I'm not sure you should use this on IPv4 - e.g. you could return the original probe on IPv4.

Nothing wrong with being specific here; always setting the IP of the host is probably a good thing (but I think it is important to also test the behavior when we omit the hostIP on the probe definition, so we test the 'default' argument).

Reply:
>  always setting the IP of the host is probably a good thing (but I think it is important to also test the behavior when we omit the hostIP on the probe definition, so we test the 'default' argument).

so I left it as it is, and added another case where no ip family is specified. in that case we won't configure the probe with host IP.
What do you think?

Reply:
@maiqueb  the default (no hostIp) [doesn't seem](https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/4341/pull-kubevirt-e2e-k8s-1.18/1321443971087470592#1:build-log.txt%3A5281) to work.

doing this locally gives:
``` 
Warning  Unhealthy  85s (x13 over 2m25s)  kubelet, node01    Readiness probe failed: dial tcp 10.244.196.158:1500: connect: connection refused
```
event in the virt-launcher pod. Can you see what I did wrong?

Reply:
When you don't set the ipFamily in the service, it defaults to the pod IP address.

Since the server is now located in a sidekick pod, the probes will not work (the server is in the sidekick pod, and kubelet is attempting to contact the VM via the virt-launcher IP address).

You should start the server on the VMI on an empty IPFamily (not creating the sidekick pod).

Reply:
Understand. will do.

What confused me is that I couldn't accept the fact that the vmi is not being checked. now that I know that it is on purpose, it is a lot more clear. Thanks for explaining.

Reply:
Done.